NSURL.m

```diff
@@ -156,6 +157,19 @@ static void posixError(CFErrorRef *error) {
     CFRelease(err);
 }
 
+static void dlError(CFErrorRef *error){
+    const CFStringRef keys[1] = {
+        kCFErrorUnderlyingErrorKey,
+    };
+    
+    CFStringRef err = CFStringCreateWithFormat(kCFAllocatorDefault, NULL, CFSTR("%s"), dlerror());
+    CFTypeRef values[1] = {
+        err,
+    };
+
+    *error = CFErrorCreateWithUserInfoKeysAndValues(kCFAllocatorDefault, kCFErrorDomainPOSIX, errno, (const void *const *)keys, (const void *const *)values, 1);
```

Comment:
Unfortunately by this point you have already clobbered `errno` (by those allocating calls above). So if you want to pass it here, save it first thing in this function (`int saved_errno = errno;`) and use that here.